### PR TITLE
WIP: Nix/feat/blockdb UI

### PR DESCRIPTION
**Pull request by DavidNix**
## Description

This PR ...

## Checklist

- [ ] I have made sure the upstream branch for this PR is correct
- [ ] I have made sure this PR is ready to merge
- [ ] I have made sure that I have assigned reviewers related to this project

## Changes

- [ ] Added ...
- [ ] Changed ...
- [ ] Removed ...

## Screenshots

...

## Testing

- Open page ...
- Click ...
- Make sure that ...

## Links/References

- ...
- ...
- ...

## Notes

- ...
- ...
- ...

### CODE REVIEW

**Comment by DavidNix** [comment]
@strangelove-ventures/ibctest-maintainers 

I'm avoid the [sunk cost fallacy](https://en.wikipedia.org/wiki/Sunk_cost) and scrap most of this in favor of using another terminal UI (TUI) framework. I also love Mark's idea of leveraging sql views. 

TLDR: I should've failed faster. 

I chose https://charm.sh because 1) great marketing 2) they're VC backed 3) includes state management

Charm/bubbletea's blockers:
* No layout primitives (like flex view). Each UI component has a different API when it comes to layout.
* No official table UI component. Tables are very useful for displaying data.
* Poor abstractions. UI components don't implement the `tea.Model` interface without an adapter. 
* Bubbletea mixes value and pointer receivers which is confusing and poor package design. 
* Using values (not pointers) of bubbletea UI components complicates updating the model.

I'm going to redo this with [tview](https://github.com/rivo/tview) of which I've used before with success. The main drawback is there's no state management for tview. (In the past, I wrote a redux-like library for managing state.)

Here is a demo of what almost was. As you can see, the layout is buggy on the last screen.

ibctest-tui1

## Roadmap

* Basic view as shown above.
* Filter to highlight values in the JSON.
* A query interface that lets you hone in on values using JSONPath or jq-like query dialect.
* Open to any ideas!

**Comment by DavidNix** [comment]
Fyi, first (but failed) attempt at https://github.com/strangelove-ventures/ibctest/issues/123
